cmd/syncthing/gui_auth.go

```diff
@@ -110,43 +116,62 @@ func basicAuthAndSessionMiddleware(cookieName string, cfg config.GUIConfiguratio
 	})
 }
 
-func AuthSimple(username string, password string, configUser string, configPassword string) (bool, string) {
-
-    if username != configUser {
-        username = string(iso88591ToUTF8([]byte(username)))
-        if username != configUser {
-            return false, ""
-        }
-    }
-
-    configPasswordBytes := []byte(configPassword)
-    passwordBytes := []byte(password)
-    if err := bcrypt.CompareHashAndPassword(configPasswordBytes, passwordBytes); err != nil {
-        passwordBytes = iso88591ToUTF8(passwordBytes)
-        if err := bcrypt.CompareHashAndPassword(configPasswordBytes, passwordBytes); err != nil {
-            return false, ""
-        }
-    }
-
-    return true, username
+func auth(username string, password string, cfg config.GUIConfiguration) bool {
+	if cfg.IsAuthModeLDAP() {
+		return AuthLDAP(username, password, cfg)
+	} else {
+		return AuthSimple(username, password, cfg.User, cfg.Password)
+	}
+}
+
+func AuthSimple(username string, password string, configUser string, configPassword string) bool {
+
+	if username != configUser {
+		return false
+	}
+
+	configPasswordBytes := []byte(configPassword)
+	passwordBytes := []byte(password)
+	if err := bcrypt.CompareHashAndPassword(configPasswordBytes, passwordBytes); err != nil {
+		return false
+	}
+
+	return true
 }
 
-func AuthLdap(username string, password string, server string, port int, bindDn string) bool {
-    l, err := ldap.Dial("tcp", fmt.Sprintf( "%s:%d", server, port))
-    if err != nil {
-        log.Println(err)
-        return false
-    }
+func AuthLDAP(username string, password string, cfg config.GUIConfiguration) bool {
+
+	address := fmt.Sprintf( "%s:%d", cfg.LDAPServer, cfg.LDAPPort)
+	var connection *ldap.Conn
+	var err error
+	if cfg.IsLDAPTLSModeTSL() {
+		connection, err = ldap.DialTLS("tcp", address, &tls.Config{InsecureSkipVerify: cfg.LDAPInsecureSkipVerify})
+	} else {
+		connection, err = ldap.Dial("tcp", address)
+	}
 
-    defer l.Close()
+	if err != nil {
+		log.Println(err)
+		return false
+	}
 
-    err = l.Bind(fmt.Sprintf(bindDn, username), password)
-    if err != nil {
-        log.Println(err)
-        return false
-    }
+	if cfg.IsLDAPTLSModeStartTSL() {
```

Comment:
Instead of this IsModeX IsModeY I'd do
```
switch cfg.GetTLSMode() {
   case TlsModeFoo:
        connection = foo.connect()
   case TlsModeBar:
         connection = barr.dial() 
```

then you only need one accessor on the config object.

Reply:
It is actually:
Plain:
  dial
Tls:
  dialTls
StartTls:
  dial
  startTls

I can do it like this if you want, just dial will be repeated :)